packages/rulesets/src/spectral/functions/query-parameters-in-collection-get.ts

```diff
@@ -0,0 +1,33 @@
+import { isListOperation } from "../../native/utilities/rules-helper"
+
+export const queryParametersInCollectionGet = (pathItem: any, _opts: any, ctx: any) => {
+  if (pathItem === null || typeof pathItem !== "object") {
+    return []
+  }
+
+  const path = ctx.path || []
+  const uris = Object.keys(pathItem)
+  if (uris.length < 1) {
+    return []
+  }
+  const GET = "get"
+  const errors: any[] = []
+
+  for (const uri of uris) {
+    //check if GET op is defined & the GET op is a collection get/list call
+    if (pathItem[uri][GET] && isListOperation(uri)) {
```

Comment:
Good that you are checking for `get` here but `isListOperation` should probably be checking if it's a `get` operation. Would it be possible to add the check there? Or if it's too much effort this is fine for now. I haven't checked how many times `isListOperation` is used throughout the code

Reply:
Thats a good point.
Lemme update.